src/libs/ReportUtils.ts

```diff
@@ -1688,7 +1695,8 @@ function getMoneyRequestReportName(report: OnyxEntry<Report>, policy: OnyxEntry<
         return Localize.translateLocal('iou.payerSpentAmount', {payer: payerName, amount: formattedAmount});
     }
 
-    if (!!report?.hasOutstandingIOU || moneyRequestTotal === 0) {
+    // The report name should show as owed also for drafts
+    if (!!report?.hasOutstandingIOU || isReportDraft(report) || moneyRequestTotal === 0) {
```

Comment:
I feel like the `!!report?.hasOutstandingIOU` is a bad pattern we got here, but not in scope to fix in this PR